internal/licenses/manager.go

```diff
@@ -0,0 +1,160 @@
+	return &manager{
+		store:      store,
+		config:     config,
+		dataReader: dataReader,
+		verifier:   verifier,
+	}
+}
+
+func (m *manager) AddLicense(ctx context.Context, licenseFilePath string, licenseKey string, publicKey string) error {
+	slog.Debug("Starting to add a new license", "filePath", licenseFilePath)
+
+	cert, err := m.dataReader(licenseFilePath)
+	if err != nil {
+		slog.Error("failed to read license file", "filePath", licenseFilePath, "error", err)
+		return fmt.Errorf("failed to read license file: %w", err)
+	}
+
+	slog.Debug("Successfully read the license file", "filePath", licenseFilePath)
+
+	lic := m.verifier(cert)
+	keygen.PublicKey = publicKey
+
+	if err := lic.Verify(); err != nil {
+		slog.Error("license verification failed", "filePath", licenseFilePath, "error", err)
+		return fmt.Errorf("license verification failed: %w", err)
+	}
+
+	dec, err := lic.Decrypt(licenseKey)
+	if err != nil {
+		slog.Error("license decryption failed", "filePath", licenseFilePath, "error", err)
+		return fmt.Errorf("license decryption failed: %w", err)
+	}
+
+	id := dec.License.ID
+	key := dec.License.Key
+
+	if err := m.store.InsertLicense(ctx, id, cert, key); err != nil {
+		slog.Error("failed to insert license into store", "licenseID", id, "error", err)
+		return fmt.Errorf("failed to insert license: %w", err)
+	}
+
+	if m.config.EnabledAudit {
+		if err := m.store.InsertAuditLog(ctx, "add", "license", id); err != nil {
+			slog.Error("failed to insert audit log", "licenseID", id, "error", err)
+			return fmt.Errorf("failed to insert audit log: %w", err)
+		}
+	}
+
+	slog.Info("added license", "licenseID", id)
+	return nil
+}
+
+func (m *manager) RemoveLicense(ctx context.Context, id string) error {
+	slog.Debug("Starting to remove license", "id", id)
+
+	if err := m.store.DeleteLicenseByID(ctx, id); err != nil {
+		slog.Error("failed to remove license", "licenseID", id, "error", err)
+		return fmt.Errorf("failed to remove license: %w", err)
+	}
+
+	if m.config.EnabledAudit {
+		if err := m.store.InsertAuditLog(ctx, "remove", "license", id); err != nil {
+			slog.Error("failed to insert audit log", "licenseID", id, "error", err)
+			return fmt.Errorf("failed to insert audit log: %w", err)
+		}
+	}
+
+	slog.Info("removed license", "licenseID", id)
+	return nil
+}
+
+func (m *manager) ListLicenses(ctx context.Context) ([]License, error) {
+	slog.Debug("Fetching licenses")
+
+	licenses, err := m.store.GetAllLicenses(ctx)
+	if err != nil {
+		slog.Error("failed to fetch licenses", "error", err)
+		return nil, err
+	}
+
+	slog.Info("fetched licenses", "count", len(licenses))
+	return licenses, nil
+}
+
+func (m *manager) GetLicenseByID(ctx context.Context, id string) (License, error) {
+	slog.Debug("Fetching license", "id", id)
+
+	license, err := m.store.GetLicenseByID(ctx, id)
+	if err != nil {
+		slog.Error("failed to fetch license by ID", "licenseID", id, "error", err)
```

Comment:
This isn't the error message that actually gets printed. What gets printed is a bit less useful:

```
Error: sql: no rows in result set
```

Reply:
Thank you!
 I have fixed the error
Now we will get something like that "license with ID license_id not found"